PROPOSAL: Update Code of Conduct to adapt Contributor Covenant 3.0 (#254)
beeware/.github: CODE_OF_CONDUCT.md

  4. **Sexualization.** Behaving in a way that would generally be considered inappropriately intimate in the context or purpose of the community.
  5. **Violating confidentiality**. Sharing or acting on someone's personal or private information without their permission.
  6. **Endangerment.** Causing, encouraging, or threatening violence or other harm toward any person or group.
  7. Behaving in other ways that **threaten the well-being** of our community.

### Other restrictions <a name="other-restrictions"></a>

  1. **Misleading identity.** Impersonating someone else for any reason, or pretending to be someone else to evade enforcement actions.
  2. **Misrepresenting project affiliation.** Speaking or acting in a way that implies an official affiliation with the BeeWare project, where one does not exist.
  3. **Failing to credit sources.** Not properly crediting the sources of content you contribute.
  4. **Promotional materials.** Sharing marketing or other commercial content in a way that is outside the norms of the community.
  5. **Excessive communication.** Disrespecting the time and space of others by engaging in an unacceptable volume of communication.

>   6. **Unnecessary interactions.** Unnecessarily reacting to or involving yourself in interactions that do not require a response from you.

Reviewer: I'm a little hesitant about this one. We don't want to discourage people from offering *informed* opinions when a subject is being discussed by others - especially when there's a power dynamic between the people discussing, and the person offering the suggestion. Suggestion:
```suggestion
  6. **Unhelpful communication.** Offering opinions without relevant experience in the topic being discussed, entering into an ongoing discussion without first gaining familiarity with the history of the topic, or making contributions that are off-topic or otherwise distracting.
```

Author: This is an excellent suggestion. I feel it is clearer and speaks more to the point I was trying to get across.

With these considerations in mind, we agree to behave mindfully toward each other and act in ways that center our shared values, including:

  1. Respecting the **purpose of our community**, our activities, and our ways of gathering.
  2. Engaging **kindly and honestly** with others.
  3. Respecting **different viewpoints** and experiences.
  4. **Taking responsibility** for our actions and contributions.
  5. Gracefully giving and accepting **constructive feedback**.
  6. Committing to **repairing harm** when it occurs.
  7. Behaving in other ways that promote and sustain the **well-being of our community**.

## Restricted behaviors <a name="restricted-behaviors"></a>

We agree to restrict the following behaviors in our community. Instances, threats, and promotion of these behaviors are violations of this Code of Conduct.

  1. **Harassment.** Violating explicitly expressed boundaries or engaging in unnecessary personal attention after any clear request to stop.
  2. **Character attacks.** Making insulting, demeaning, or pejorative comments directed at a community member or group of people.

>   3. **Stereotyping or discrimination.** Characterizing anyone’s personality or behavior on the basis of immutable identities or traits.

Reviewer: That's a fair comment - I'm trying to think if we open up any issues if we drop "immutable" from this phrasing, and I can't see any issues... can anyone else see any problems?
```suggestion
  3. **Stereotyping or discrimination.** Characterizing anyone’s personality or behavior on the basis of a personal identity or trait.
```

>   3. **Stereotyping or discrimination.** Characterizing anyone’s personality or behavior on the basis of immutable identities or traits.

Reviewer 2: This may be splitting hairs at this point, as the title "stereotyping or discrimination" makes the intent pretty clear... but literally speaking, "characterizing a person's personality or behavior on the basis of a trait" can describe basically any instance of describing what someone's like. Like "He greets everyone, because he's really friendly" fits that definition.

  4. **Sexualization.** Behaving in a way that would generally be considered inappropriately intimate in the context or purpose of the community.
  5. **Violating confidentiality.** Sharing or acting on someone's personal or private information without their permission.
  6. **Endangerment.** Causing, encouraging, or threatening violence or other harm toward any person or group.
  7. Behaving in other ways that **threaten the well-being** of our community.

### Other restrictions <a name="other-restrictions"></a>

  1. **Misleading identity.** Impersonating someone else for any reason, or pretending to be someone else to evade enforcement actions.
  2. **Misrepresenting project affiliation.** Speaking or acting in a way that implies an official affiliation with the BeeWare project, where one does not exist.

>   3. **Failing to credit sources.** Not properly crediting the sources of content you contribute.

Reviewer: The intention (as I read it) is directed more at plagiarism than "quote attribution"; but the latter *can* be a factor if it's egregious. 

Take the example of someone who presents a really good metaphor to describe a complex problem, and someone else takes that metaphor and presents it as their own, without attributing the originating idea. It's not strictly a *copyright* issue, as an *idea* isn't copyrightable; but there's definite prior work that isn't getting recognition. Another example is a manager who takes sole credit for work done by juniors.

>   3. **Failing to credit sources.** Not properly crediting the sources of content you contribute.

Reviewer 3: Ah okay, "give due credit when using other people's ideas" absolutely makes sense. But from the wording "crediting the sources of content you contribute", that wasn't really clear to me. The "crediting sources" part made me think of quotes and the "content you contribute" part made me think of copyright/license issues. I think this could be worded more clearly (no concrete suggestions from me though).

>   3. **Failing to credit sources.** Not properly crediting the sources of content you contribute.

Reviewer: I think I can see where you're coming from, but I'm not sure I agree that it's sufficiently unclear to pose a problem. It's fairly loose as a definition (anything could be "content you contribute"), and "credit" doesn't have to mean "citations", which gives the CoC team leeway to consider fairly broad context for responding to a complaint.

  7. **Irresponsible messaging.** Failing to responsibly present content which includes, links or describes any other restricted behaviors.
  8. Other conduct that could reasonably be considered **unprofessional or inappropriate**.

## Reporting an issue <a name="reporting-an-issue"></a>

Tensions can occur between community members even when they are trying their best to collaborate. Not every conflict represents a code of conduct violation, and this Code of Conduct reinforces encouraged behaviors and norms that can help avoid conflicts and minimize harm. Reporting even minor issues is important, as they can be helpful in identifying patterns of behavior that may not be concerning in isolation, but when viewed collectively may be more significant.

When an incident does occur, it is important to report it promptly to the BeeWare Code of Conduct Response Team.

**If you believe you or anyone else is in physical danger, please notify appropriate law enforcement first.**

> To report a possible violation, email the Team at [conduct@beeware.org](mailto:conduct@beeware.org). If necessary, you can reach out to individual team members. On the BeeWare Discord server, you can also direct message anyone on the Response Team, or, if appropriate, mention @moderators in a public channel. Team members can be reached by the following usernames on Discord or GitHub, or the provided email addresses:

Reviewer 3: GitHub's syntax highlighting implies that `@moderators` might be parsed as a user mention, so escape it just in case...

```suggestion
To report a possible violation, email the Team at [conduct@beeware.org](mailto:conduct@beeware.org). If necessary, you can reach out to individual team members. On the BeeWare Discord server, you can also direct message anyone on the Response Team, or, if appropriate, mention \@moderators in a public channel. Team members can be reached by the following usernames on Discord or GitHub, or the provided email addresses:
```

  * Russell Keith-Magee (@freakboy3742; [russell@beeware.org](mailto:russell@beeware.org))
  * Kattni (@kattni; [kattni@beeware.org](mailto:kattni@beeware.org))
  * Katie McLaughlin (@glasnt; [katie@beeware.org](mailto:katie@beeware.org))
  * Philip James (@phildini; [philip@beeware.org](mailto:philip@beeware.org))
  * Charles Whittington (@HalfWhitt; [charles@beeware.org](mailto:charles@beeware.org))

> The Response Team takes reports of violations seriously and will make every effort to respond in a timely manner. They will investigate all reports of code of conduct violations, reviewing messages, logs, and recordings, or interviewing witnesses and other participants. The Team will keep investigation and enforcement actions as transparent as possible while prioritizing safety and confidentiality. In order to honor these values, enforcement actions are carried out in private with the involved parties, but communicating to the whole community may be part of a mutually agreed upon resolution. If we determine that a public statement will need to be made, the identities of all victims and reporters will remain confidential unless those individuals instruct us otherwise.

Reviewer 2: ```suggestion
The Response Team takes reports of violations seriously and will make every effort to respond in a timely manner. They will investigate all reports of code of conduct violations, reviewing messages, logs, and recordings, or interviewing witnesses and other participants. The Team will keep investigation and enforcement actions as transparent as possible while prioritizing safety and confidentiality. In order to honor these values, enforcement actions are carried out in private with the involved parties, but communicating to the whole community may be part of a mutually agreed upon resolution. If we determine that a public statement needs to be made, the identities of all victims and reporters will remain confidential unless those individuals instruct us otherwise.
```

     * Repair: Examples of repair include respecting the spirit of the suspension, meeting the specified conditions for return, and being thoughtful about how to reintegrate with the community when the suspension is lifted.

  4. Permanent Ban
     * Event: A pattern of repeated code of conduct violations that other steps on the ladder have failed to resolve, or a violation so serious that the Response Team determines there is no way to keep the community safe with this person as a member.
     * Consequence: Access to all community spaces, tools, and communication channels is removed. In general, permanent bans should be rarely used, should have strong reasoning behind them, and should only be resorted to if working through other remedies has failed to change the behavior.
     * Repair: There is no possible repair in cases of this severity.

This enforcement ladder is intended as a guideline. It does not limit the ability of Community Managers to use their discretion and judgment, in keeping with the best interests of our community.

> ## Scope <a name="scope"></a>

Reviewer: What sort of interaction are you considering here? That BeeWare's CoC "yes-and"s the CoC of an event? Or more explicit details about how BeeWare's CoC team will work with conference CoC teams?

> ## Scope <a name="scope"></a>

Reviewer 4: tbc, this is a minor thing, and we can choose to do nothing

> ## Scope <a name="scope"></a>

Reviewer: A fair point; probably worth a note pointing out that:
* multiple CoC may be in effect at some times;
* we'd encourage people to report an incident to all relevant groups;
* Known violation of other CoC may be considered as evidence in a BeeWare CoC complaint.
* that the BeeWare CoC group won't share information with other CoC groups without consent of the reporting party.

This Code of Conduct applies within all community spaces, including GitHub, the BeeWare Discord server, and in-person events, such as conferences, meetups, and sprints. It also applies when an individual is officially representing the community in public or other spaces. Examples of representing our community include using an official email address, posting via an official social media account, or acting as an appointed representative at an online or offline event.

> Behavior outside of official BeeWare spaces may also be considered as supporting evidence for a report if that behavior establishes a pattern, or represents a potential risk to the BeeWare community. 

Reviewer: pre-commit whitespace issue:
```suggestion
Behavior outside of official BeeWare spaces may also be considered as supporting evidence for a report if that behavior establishes a pattern, or represents a potential risk to the BeeWare community.
```
